src/internal/utils/io.go

```diff
@@ -94,31 +97,31 @@ func ReplaceTextTemplate(path string, mappings map[string]string) {
 }
 
 // RecursiveFileList walks a path with an optional regex pattern and returns a slice of file paths
-func RecursiveFileList(root string, pattern *regexp.Regexp) []string {
-	var files []string
+func RecursiveFileList(dir string, pattern *regexp.Regexp) (files []string, err error) {
+	err = filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		// Skip hidden directories
+		if d.IsDir() && d.Name()[0] == dotCharacter {
```

Comment:
just FYSA this isn't going to work for [Windows](https://freshman.tech/snippets/go/detect-hidden-file/#detect-hidden-files-and-directories-in-windows).

TLDR; Windows has an explicit file attribute for hidden and dots at the beginning don't matter.

Reply:
Yeah thanks I was actually tracking that, but figured since we don't have windows support yet it was a bit of a moot point.